esrally/storage/_adapter.py

```diff
@@ -105,7 +90,7 @@ def match_url(cls, url: str) -> bool:
         raise NotImplementedError
 
     @classmethod
-    def from_config(cls, cfg: Config) -> Adapter:
+    def from_config(cls: type[A], cfg: Config, **kwargs: dict[str, Any]) -> A:
```

Comment:
I think `typing_extensions` has a `Self` type that works exactly like this for any class. So you can just write `def from_config(cls, cfg: Config, **kwargs: Any) -> Self:`

Also, `**kwargs` like this is typed by the value type only. [Reference](https://peps.python.org/pep-0484/#arbitrary-argument-lists-and-default-argument-values)